sdl_viewer/src/terrain_drawer/read_write.rs

```diff
@@ -0,0 +1,56 @@
+use crate::graphic::tiled_texture_loader::{TilePos, TiledTextureLoader};
+use cgmath::Vector3;
+use image::{LumaA, Rgba};
+use point_viewer::math::Isometry3;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::{Error, ErrorKind};
+
+#[derive(Serialize, Deserialize)]
+pub struct Metadata {
+    pub tile_size: u32,
+    pub world_from_terrain: Isometry3<f64>,
+    pub origin: Vector3<f64>,
+    pub resolution_m: f64,
+    pub tile_positions: Vec<TilePos>,
+}
+
+pub type TextureLoaders = (TiledTextureLoader<LumaA<f32>>, TiledTextureLoader<Rgba<u8>>);
+
+#[allow(dead_code)]
+impl Metadata {
+    // Custom serialization of some values
+    pub fn from_dir<P: AsRef<std::path::Path>>(dir: P) -> Result<Self, Error> {
+        let meta_path = dir.as_ref().join("meta.json");
+        let reader = File::open(meta_path)?;
+        let meta = serde_json::from_reader(reader).map_err(|e| {
+            let msg = format!("Could not parse meta.json: {}", e.to_string());
+            Error::new(ErrorKind::InvalidData, msg)
+        })?;
+        Ok(meta)
+    }
+
+    pub fn read_tiles<P: AsRef<std::path::Path>>(&self, dir: P) -> Result<TextureLoaders, Error> {
+        let height_tiles = self.tile_positions.iter().map(|xy| {
```

Comment:
nit: do `.map(|(x, y)| { ... ` instead of `.map(|xy| { ...` so we don't need line `35`?

Reply:
Done